### PR TITLE
Feature/ebp 100

### PR DESCRIPTION
**Pull request by AlKaliada**

### CODE REVIEW

File: core/src/main/java/com/exadel/etoolbox/backpack/core/services/pckg/impl/BasePackageServiceImpl.java [hunk 1]
```diff
@@ -155,7 +159,8 @@ public PackageInfo getPackageInfo(final ResourceResolver resourceResolver, final
         } else {
             actualPaths = packageModel.getPaths().stream()
                     .filter(s -> resourceResolver.getResource(s.getPath()) != null)
-                    .map(path -> getActualPath(path.getPath(), path.isExcludeChildren(), resourceResolver))
+                    .flatMap(pathModel -> liveCopyService.getResourcePathWithLiveCopiesPaths(resourceResolver, pathModel.getPath(), pathModel.isIncludeLiveCopies())
```
Comment (smiakchilo) [inline core/src/main/java/com/exadel/etoolbox/backpack/core/services/pckg/impl/BasePackageServiceImpl.java:162]: (cosmetic) Consider moving the part after `pathModel ->` to a separate method. It is a Stream API chain within a Stream API chain == always reads with difficulty

File: core/src/main/java/com/exadel/etoolbox/backpack/core/servlets/model/PathModel.java [hunk 2]
```diff
@@ -52,9 +56,17 @@ public String getPath() {
 
     /**
      * Gets the whether or not child pages excluded from a package build
```
Comment (smiakchilo) [inline core/src/main/java/com/exadel/etoolbox/backpack/core/servlets/model/PathModel.java:58]: Replace "excluded" with "included" in the Javadoc. BTW, not just "included" but "are included")

File: core/src/main/java/com/exadel/etoolbox/backpack/core/servlets/model/PathModel.java [hunk 3]
```diff
@@ -52,9 +56,17 @@ public String getPath() {
...
-     * @return String value
+     * @return boolean value
+     */
+    public boolean isIncludeChildren() {
```
Comment (smiakchilo) [inline core/src/main/java/com/exadel/etoolbox/backpack/core/servlets/model/PathModel.java:61]: (naming) Why "isInclude"? You are not obliged to add "is..." to whatever Boolean-typed getter. Code must be read as text, and "is include something" is a grammatical error.

File: ui.apps/src/main/content/jcr_root/apps/etoolbox-backpack/admin/console/page/content/clientlibs/js/paths.js [hunk 4]
```diff
@@ -43,4 +44,30 @@ $(function () {
...
+                    prefix += '-multiple';
+                }
+                name.set('value', prefix);
+            }
+        }, 100);
+    }
+
+    function longestCommonPrefix(strings) {
+        var sorted = strings.sort((a, b) => a.length - b.length);
+        var shorted = sorted[0];
```
Comment (smiakchilo) [inline ui.apps/src/main/content/jcr_root/apps/etoolbox-backpack/admin/console/page/content/clientlibs/js/paths.js:64]: What does it mean, "shorted"?)

File: core/src/main/java/com/exadel/etoolbox/backpack/core/services/impl/LiveCopyServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+@Component(service = LiveCopyService.class)
+public class LiveCopyServiceImpl implements LiveCopyService {
+
+    @Reference
+    private LiveRelationshipManager liveRelationshipManager;
+
+    @Override
+    public List<String> getResourcePathWithLiveCopiesPaths(ResourceResolver resourceResolver, String path, boolean includeLiveCopies) {
+        List<String> paths = new ArrayList<>(Arrays.asList(path));
```
[the remark this answers is not in the document]
Comment (AlKaliada) [reply core/src/main/java/com/exadel/etoolbox/backpack/core/services/impl/LiveCopyServiceImpl.java:26]: done

File: core/src/main/java/com/exadel/etoolbox/backpack/core/services/impl/LiveCopyServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+                LiveCopy liveCopy = relationship.getLiveCopy();
+                if (liveCopy == null) {
+                    continue;
+                }
+                String liveCopyPath = liveCopy.getPath() + relationship.getSyncPath();
+                if (resourceResolver.getResource(liveCopyPath) != null) {
+                    paths.add(liveCopyPath);
+                }
+            }
+        } catch (WCMException e) {
```
[the remark this answers is not in the document]
Comment (AlKaliada) [reply core/src/main/java/com/exadel/etoolbox/backpack/core/services/impl/LiveCopyServiceImpl.java:44]: sorry, forgot about it :(
done

File: core/src/main/java/com/exadel/etoolbox/backpack/core/services/pckg/impl/BasePackageServiceImpl.java [hunk 7]
```diff
@@ -99,6 +100,9 @@ public class BasePackageServiceImpl implements BasePackageService {
     @Reference
     protected QueryService queryService;
 
+    @Reference
+    protected LiveCopyService liveCopyService;
```
[the remark this answers is not in the document]
Comment (AlKaliada) [reply core/src/main/java/com/exadel/etoolbox/backpack/core/services/pckg/impl/BasePackageServiceImpl.java:104]: done

File: core/src/main/java/com/exadel/etoolbox/backpack/core/services/pckg/impl/BasePackageServiceImpl.java (hunk 1 above)
Comment (AlKaliada) [reply core/src/main/java/com/exadel/etoolbox/backpack/core/services/pckg/impl/BasePackageServiceImpl.java:162]: done

File: core/src/main/java/com/exadel/etoolbox/backpack/core/servlets/model/PathModel.java (hunk 3 above)
Comment (AlKaliada) [reply core/src/main/java/com/exadel/etoolbox/backpack/core/servlets/model/PathModel.java:61]: done

File: core/src/main/java/com/exadel/etoolbox/backpack/core/servlets/model/PathModel.java (hunk 2 above)
Comment (AlKaliada) [reply core/src/main/java/com/exadel/etoolbox/backpack/core/servlets/model/PathModel.java:58]: done

File: ui.apps/src/main/content/jcr_root/apps/etoolbox-backpack/admin/console/page/content/clientlibs/js/paths.js (hunk 4 above)
Comment (AlKaliada) [reply ui.apps/src/main/content/jcr_root/apps/etoolbox-backpack/admin/console/page/content/clientlibs/js/paths.js:64]: shortest :)

File: core/src/main/java/com/exadel/etoolbox/backpack/core/services/LiveCopyService.java [hunk 8]
```diff
@@ -1,11 +1,10 @@
...
 
-import org.apache.sling.api.resource.Resource;
 import org.apache.sling.api.resource.ResourceResolver;
 
 import java.util.List;
 
 public interface LiveCopyService {
 
-    List<String> getResourcePathWithLiveCopiesPaths(ResourceResolver resourceResolver, java.lang.String path, boolean includeLiveCopies);
+    List<String> getPaths(ResourceResolver resourceResolver, java.lang.String path, boolean includeLiveCopies);
```
Comment (smiakchilo) [inline core/src/main/java/com/exadel/etoolbox/backpack/core/services/LiveCopyService.java:9]: Pls remove `lava.lang.` from `String`

File: ui.apps/src/main/content/jcr_root/apps/etoolbox-backpack/admin/console/page/content/clientlibs/js/paths.js [hunk 9]
```diff
@@ -14,8 +14,40 @@
 $(function () {
     "use strict";
 
+    $(document).on("click", '.js-create-package-activator', function(event) {
+        var dialog = document.querySelector('#createPackageDialog');
+        if (!dialog) {
+            var path = $('.js-create-package-activator').data('path');
+            $.ajax({
+                type: "GET",
```
Comment (smiakchilo) [inline ui.apps/src/main/content/jcr_root/apps/etoolbox-backpack/admin/console/page/content/clientlibs/js/paths.js:22]: (cosmetic) Pls unify quotes. Use single quotes all around

File: ui.apps/src/main/content/jcr_root/apps/etoolbox-backpack/admin/console/page/content/clientlibs/js/paths.js [hunk 10]
```diff
@@ -14,8 +14,40 @@
...
+                    var doc = new DOMParser().parseFromString(data, 'text/html');
+                    document.body.appendChild(doc.body.firstChild);
+                }
+            });
+        }
+        setTimeout(function() {
+            dialog = document.querySelector('#createPackageDialog');
+            fillPaths();
+            dialog.show();
+        }, 300);
```
Comment (smiakchilo) [inline ui.apps/src/main/content/jcr_root/apps/etoolbox-backpack/admin/console/page/content/clientlibs/js/paths.js:34]: BTW, what this timeout is for? It used to be 100 yesterday, now it's 300.

I noticed that you query for the same object twice: first in line 18, and second, in line 31. 
I may be mistaken but it looks like you are not sure that some node is already there so you just sit and wait for it for an arbitrary period of time (which may vary depending on the end-user's machine). 

You may want to use instead [Coral.commons.ready](https://developer.adobe.com/experience-manager/reference-materials/6-5/coral-ui/coralui3/Coral.commons.html#Coral.commons:.ready) (if waiting for a Coral component)

or [dialog-ready](https://stackoverflow.com/questions/49779675/dialog-ready-event-does-not-fire-when-dialog-opens-in-aem-6-1)

or [foundation-contentloaded](https://developer.adobe.com/experience-manager/reference-materials/6-5/granite-ui/api/jcr_root/libs/granite/ui/components/foundation/clientlibs/foundation/vocabulary/contentloaded.html)

Timer is a last resort.

File: ui.apps/src/main/content/jcr_root/apps/etoolbox-backpack/admin/console/page/content/clientlibs/js/paths.js [hunk 11]
```diff
@@ -14,8 +14,40 @@
...
+    });
+
     $(document).on("foundation-contentloaded", function() {
 
+        fillPaths();
+
+        var switchField = $('#create-package-switch')[0];
```
Comment (smiakchilo) [inline ui.apps/src/main/content/jcr_root/apps/etoolbox-backpack/admin/console/page/content/clientlibs/js/paths.js:41]: Look, you do a contradictory thing:
1) init a JQuery object with `$('#create-package-switch')`
2) then de-structure it into a DOM object with `[0]`
3) then re-init all the same JQuery object with `$(switchField)`

Just do `$('#create-package-switch').on('change'...`

File: ui.apps/src/main/content/jcr_root/apps/wcm/core/content/editor/_jcr_content/content/items/content.xml [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+                <items jcr:primaryType="nt:unstructured">
+                    <pageinfopopover
+                        jcr:primaryType="nt:unstructured">
+                        <items jcr:primaryType="nt:unstructured">
+                            <list
+                                jcr:primaryType="nt:unstructured">
+                                <items jcr:primaryType="nt:unstructured">
+                                    <backpack
+                                            granite:class="js-create-package-activator"
+                                            granite:title="Pack by Backpack"
```
Comment (smiakchilo) [inline ui.apps/src/main/content/jcr_root/apps/wcm/core/content/editor/_jcr_content/content/items/content.xml:18]: My vote is *still* for `Create package` + BackPack icon

**Pull request merged**